### ISSUE TITLE
error: This version of ChromeDriver only supports Chrome version 110

### ISSUE THREAD

**Issue by sergerdn** [CheshireCaat/selenium-with-fingerprints#1]
Please note that I'm not an experienced programmer with JavaScript, so there's a possibility that I may be missing something.
I created a simple script based on the documentation you provided.

```javascript
const { plugin } = require('selenium-with-fingerprints');

(async () => {
  // Get a fingerprint from the server:
  const fingerprint = await plugin.fetch('', {
    tags: ['Microsoft Windows', 'Chrome'],
  });

  // Apply fingerprint:
  plugin.useFingerprint(fingerprint);

  // Launch the browser instance:
  const driver = await plugin.launch();

  // The rest of the code is the same as for a standard `selenium` library:
  await driver.get('https://example.com');

  // Print the browser viewport size:
  console.log(
    'Viewport:',
    await driver.executeScript(() => ({
      deviceScaleFactor: window.devicePixelRatio,
      width: document.documentElement.clientWidth,
      height: document.documentElement.clientHeight,
    }))
  );

  await driver.quit();
})();

```

However, when I tried to execute the example provided, I encountered errors instead of a successful run.

```
"C:\Program Files\nodejs\node.exe" C:\Users\Administrator\WebstormProjects\selenium-with-fingerprints-testing\example1.js
Debugger listening on ws://127.0.0.1:49836/f280d756-d05c-4447-a08a-dd607da8cc82                                         
Dor help, see: https://nodejs.org/en/docs/inspector                                                                     
Debugger attached.                                                                                                      
                                                                                                                        
The browser is downloading - this may take some time.                                                                   
                                                                                                                        
                                                                                                                        
Debugger listening on ws://127.0.0.1:49836/f280d756-d05c-4447-a08a-dd607da8cc82
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
The browser is downloading - this may take some time.
The browser is installing - this may take some time. 
Try the full version of the fingerprint service which has advantages over the free one:                                 
                                                                                                                        
- Emulate fingerprints for other platforms and browsers.                                                                
- Increased limits when receiving a fingerprints from the service.                                                      
- Filters by tags, versions, screen resolution and many other parameters.                                               
- Popular perfect canvas queries are already included in the received fingerprints.                                     
                                                                                                                        
If you want to buy the full version, please visit this website - https://bablosoft.com/directbuy/FingerprintSwitcher/2. 
 The ChromeDriver could not be found on the current PATH, trying Selenium Manager 

DevTools listening on ws://127.0.0.1:49903/devtools/browser/f1ec4278-bfbb-47e8-97d4-7c1a0d1383c5 
Waiting for the debugger to disconnect... 
C:\Users\Administrator\WebstormProjects\selenium-with-fingerprints-testing\node_modules\selenium-webdriver\lib\error.js:524 
    let err = new ctor(data.message)
              ^

SessionNotCreatedError: session not created: This version of ChromeDriver only supports Chrome version 110
Current browser version is 109.0.5414.75 with binary path C:/Users/Administrator/WebstormProjects/selenium-with-fingerprints-testing/data/run/FingerprintPlugin/d5219/browser/chrome/worker/worker.exe
    at Object.throwDecodedError (C:\Users\Administrator\WebstormProjects\selenium-with-fingerprints-testing\node_modules\selenium-webdriver\lib\error.js:524:15)
    at parseHttpResponse (C:\Users\Administrator\WebstormProjects\selenium-with-fingerprints-testing\node_modules\selenium-webdriver\lib\http.js:587:13)
    at Executor.execute (C:\Users\Administrator\WebstormProjects\selenium-with-fingerprints-testing\node_modules\selenium-webdriver\lib\http.js:515:28)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  remoteStacktrace: 'Backtrace:\n' +
    '\t(No symbol) [0x00D137D3]\n' +
    '\t(No symbol) [0x00CA8B81]\n' +
    '\t(No symbol) [0x00BAB36D]\n' +
    '\t(No symbol) [0x00BCED6D]\n' +
    '\t(No symbol) [0x00BC9B90]\n' +
    '\t(No symbol) [0x00BC6FC9]\n' +
    '\t(No symbol) [0x00C01ED5]\n' +
    '\t(No symbol) [0x00C01B2C]\n' +
    '\t(No symbol) [0x00BFB216]\n' +
    '\t(No symbol) [0x00BD0D97]\n' +
    '\t(No symbol) [0x00BD253D]\n' +
    '\tGetHandleVerifier [0x00F8ABF2+2510930]\n' +
    '\tGetHandleVerifier [0x00FB8EC1+2700065]\n' +
    '\tGetHandleVerifier [0x00FBC86C+2714828]\n' +
    '\tGetHandleVerifier [0x00DC3480+645344]\n' +
    '\t(No symbol) [0x00CB0FD2]\n' +
    '\t(No symbol) [0x00CB6C68]\n' +
    '\t(No symbol) [0x00CB6D4B]\n' +
    '\t(No symbol) [0x00CC0D6B]\n' +
    '\tBaseThreadInitThunk [0x751B05C9+25]\n' +
    '\tRtlGetAppContainerNamedObjectPath [0x771F784D+237]\n' +
    '\tRtlGetAppContainerNamedObjectPath [0x771F781D+189]\n'
}

Node.js v18.14.2

Process finished with exit code 1
```

**Comment by CheshireCaat** [comment]
@sergerdn, good day! For now, for the `selenium` framework you must manage the `chromedriver` version 
yourself so that it matches the browser version in the plugin, because all these packages are peer dependencies.

To simplify the task, you can find the correct version [here](https://github.com/CheshireCaat/selenium-with-fingerprints/blob/master/package.json#L53). You can also look at [examples](https://github.com/CheshireCaat/selenium-with-fingerprints/tree/master/examples) - they all use the `chromedriver` npm package to work.

**Comment by sergerdn** [comment]
Thank you for your message. Please do not close the issue yet, as it is still very confusing to me, and I have further questions regarding your answer. I have read your response multiple times and looked at your links repeatedly, but I still do not understand what I need to do exactly to fix the errors.

Should I download the Chrome driver for the corresponding version, save it somewhere, and then update the example to run with my downloaded Chrome driver? Or am I mistaken?

https://chromedriver.chromium.org/downloads

**Comment by CheshireCaat** [comment]
OK, it's just that this question is more related to selenium than to a plugin.

You can use the [chromedriver](https://www.npmjs.com/package/chromedriver) package of the version listed in the link above or in the text of your error - for now it's `109.0.0`. In scripts, you need to import it before working with `selenium`.

Another option is to install the desired version locally and update the `PATH` environment variable to include the path to the driver, just like your error says:

> The ChromeDriver **could not be found** on the current PATH, trying Selenium Manager 

The simplest is to use the `chromedriver` package - it just does all the work for you.

**Comment by sergerdn** [comment]
Thank you! I was finally able to fix it in the example project by running `npm install chromedriver@109.0.0 --chromedriver-force-download` in the root directory and  adding the line `require('chromedriver');` to the top of the example script:
```javascript

require('chromedriver');  # added this line
```

My current `package.json`:
```json
{
  "dependencies": {
    "chromedriver": "^109.0.0",
    "selenium-webdriver": "^4.8.1",
    "selenium-with-fingerprints": "^1.1.2"
  }
}
```

Would you like to update your documentation in README.md with this information? I believe it will be helpful for people to avoid such errors.

If you need it, I am able to write pull request for you with changes to the README.md file.

**Comment by CheshireCaat** [comment]
I'll add extra-info a bit later, thanks for suggestion!

**Comment by sergerdn** [comment]
> I'll add extra-info a bit later, thanks for suggestion!


Would it be okay with you if will close this issue with a commit message such as "closes issue #1"? This will help to associate the commit with the issue and provide a clear reference for future development.

For more information on associating commits with GitHub issues, you can find additional details at https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue.

In this case, I will receive an email notification from GitHub once you have fixed the issue.

**Comment by CheshireCaat** [comment]
OK, it seems reasonable.
